cmd/datasource-syncer/main.go

```diff
@@ -0,0 +1,205 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	grafana "github.com/grafana/grafana-api-golang-client"
+	"golang.org/x/mod/semver"
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/google"
+)
+
+var (
+	grafanaToken = flag.String("grafana-api-token", "",
+		"grafana-api-token used to access Grafana. Can be created using: https://grafana.com/docs/grafana/latest/administration/service-accounts/#create-a-service-account-in-grafana")
+	grafanaEndpoint    = flag.String("grafana-api-endpoint", "", "grafana-api-endpoint is the endpoint of the Grafana instance that contains the datasources to update.")
+	datasourceUIDList  = flag.String("datasource-uids", "", "datasource-uids is a comma separated list of datasource UIDs to update.")
```

Comment:
The way this is written, you can update multiple datasource IDs with the same exact prometheus URL... but that's the same metrics scope... so you could update multiple datasources at the same Grafana instance to query the same set of data, which seems kind of pointless?

Feels like we should either make it so that the datasources are paired with a project ID, or remove the array for datasource ID. Or at least de-emphasize the array and say that in almost all situations you'll just pick one ID.

You'd also have to make sure the same service account has permissions to multiple scoping projects, but that's doable, if not best practice.